packages/messaging/src/controllers/sw-types.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Subset of Web Worker types from lib.webworker.d.ts
+ *
+ * Since it's not possible to have both "dom" and "webworker" libs in a single
+ * project, we have to manually declare the web worker types we need.
+ */
+
+// Not the whole interface, just the parts we're currently using.
+// If TS claims that something does not exist on this, feel free to add it.
+interface ServiceWorkerGlobalScope {
```

Comment:
Does this not already exist as a TSLib?

Reply:
It exists (as a type definition file), but it can't be used simultaneously with DOM.

https://github.com/Microsoft/TypeScript/issues/20595
https://github.com/Microsoft/TypeScript/issues/14687

So I took some parts of it that we need (and doesn't conflict with DOM) and put them here.

Reply:
Added link.